packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java

```diff
@@ -66,37 +76,48 @@ private static void setAudioAttributes(ExoPlayer exoPlayer, boolean isMixMode) {
         !isMixMode);
   }
 
-  void play() {
+  @Override
+  public void play() {
     exoPlayer.play();
   }
 
-  void pause() {
+  @Override
+  public void pause() {
     exoPlayer.pause();
   }
 
-  void setLooping(boolean value) {
-    exoPlayer.setRepeatMode(value ? REPEAT_MODE_ALL : REPEAT_MODE_OFF);
+  @Override
+  public void setLooping(@NonNull Boolean looping) {
+    exoPlayer.setRepeatMode(looping ? REPEAT_MODE_ALL : REPEAT_MODE_OFF);
   }
 
-  void setVolume(double value) {
-    float bracketedValue = (float) Math.max(0.0, Math.min(1.0, value));
+  @Override
+  public void setVolume(@NonNull Double volume) {
+    float bracketedValue = (float) Math.max(0.0, Math.min(1.0, volume));
     exoPlayer.setVolume(bracketedValue);
   }
 
-  void setPlaybackSpeed(double value) {
+  @Override
+  public void setPlaybackSpeed(@NonNull Double speed) {
     // We do not need to consider pitch and skipSilence for now as we do not handle them and
     // therefore never diverge from the default values.
-    final PlaybackParameters playbackParameters = new PlaybackParameters(((float) value));
+    final PlaybackParameters playbackParameters = new PlaybackParameters(speed.floatValue());
 
     exoPlayer.setPlaybackParameters(playbackParameters);
   }
 
-  void seekTo(int location) {
-    exoPlayer.seekTo(location);
+  @Override
+  public @NonNull Long getPosition() {
+    long position = exoPlayer.getCurrentPosition();
+    // TODO(stuartmorgan): Move this; this is relying on the fact that getPosition is called
+    //  frequently to drive buffering updates, which is a fragile hack.
```

Comment:
This hack is pre-existing; it's just being move from the VideoPlayerPlugin.java version of this method. I added the comment so that future people don't have to go through the same process I did of wondering why getting the position sends a buffer update as a side effect.